seclang/parser.go

```diff
@@ -77,40 +76,50 @@ func (p *Parser) FromFile(profilePath string) error {
 // or arguments are invalid
 func (p *Parser) FromString(data string) error {
 	scanner := bufio.NewScanner(strings.NewReader(data))
-	var linebuffer = ""
-	pattern := regexp.MustCompile(`\\(\s+)?$`)
+	var linebuffer strings.Builder
 	inQuotes := false
 	for scanner.Scan() {
 		p.currentLine++
 		line := strings.TrimSpace(scanner.Text())
-		if !inQuotes && len(line) > 0 && line[len(line)-1] == '`' {
+		lineLen := len(line)
+		if lineLen == 0 {
+			continue
+		}
+
+		if !inQuotes && line[lineLen-1] == '`' {
```

Comment:
Still hoping for an example on this one :)

https://github.com/corazawaf/coraza/pull/398#discussion_r963182382

Reply:
It was implemented for SecDataset:

- !inQuote ensures we are not inside an action list
- When we are reading the directive, and we declare a list using "`", the value of the whole line will be "`"
- We can only close opened "`" with a single "`"

It's an ugly code, please if you have a better idea go ahead

Reply:
Ah thanks for the info! IIUC, then we could replace to just be `line == "`"` - that would clear up the confusion I had nicely

Reply:
Ok so I think then we should call it `inBackticks` instead of `inQuotes` and then, what we are trying to support is something like:

```
SecDataset test `
  123
  456
`
```

So the first condition matches the last backtick in **SecDataset test `** right? and as for the last we can simply match it with `line == "\`"` as rag suggested. 

I wonder if there is a case (cc @fzipi @M4tteoP @piyushroshan) where a backtick is at the beginning or at the end and it is not inside a `SectDataset` or a similar construct.

Reply:
I see where the len-1 comes from now. I think a `\` or a `#` may be able to break that assumption though. Let's add test cases for these both on first quote and last.

But ok with filing an issue and handling in a separate PR since it's not related to performance which this PR is handling.

Reply:
I think the backtick is a blackhole in that sense, whatever you add inside despite a keyword somewhere else (e.g. `#` for comments) lost its ability inside backticks.

Reply:
Speaking about `SecDataset`, comments (`#`) are evaluated and stripped later on, demanding it to [directiveSecDataset](https://github.com/corazawaf/coraza/blob/c7ec8112b8a94860a27663d3a1908a4fdc1f8cb8/seclang/directives.go#L487) and not to the initial parser.

Reply:
Yeah I was thinking mostly about trailing comments, especially of the ending quote. But actually I guess might not handle them at all right now. Basically a line should be 

`trim(line)[0:LastIndexByte('#')]`

type of thing